### ISSUE TITLE
How to force update check?

### ISSUE THREAD

**Issue by jampy** [NekR/offline-plugin#137]
AFAIK the plugin (or, more correctly, it's service worker) only checks for changed files (new application release) when the webpage "starts"/loads.

Is there a way (or can it be added) to manually force a check via a function call or at least define a re-check interval.

Reason: My application (PWA) is likely to remain opened (or, at least, live in the background) for a long time. Unless the user force-closes applications or restarts the phone, the application (which regularly gets updates) won't notice that a new version is available. I would like that at most 3-4 hours after the last check  the webserver is contacted again to see if a new version has been released.

**Comment by NekR** [comment]
@jampy hey, there is no such a way at a moment, but that's indeed good to have.

`offline-plugin` doesn't add special restrictions to SW or AppCache updates, so while there is no built-in function for that, you can still do that manually like this:

```js
function updateOfflineService() {
  if (navigator.serviceWorker) {
    navigator.serviceWorker.ready.then((registration) => {
      return registration.update();
    });
  } else if (applicationCache.update) {
    applicationCache.update();
  }
}
```

**Important:** Be sure to server SW file without http caching (specify explicitly to not cache), otherwise you make get updates from the cache.

**Comment by jampy** [comment]
Works fine, thanks!

**Comment by NekR** [comment]
@jampy great :-)

**Comment by jampy** [comment]
I tried to patch offline-plugin to add `updateOfflineService()` to the runtime, but there is a complicated runtime.js build behind the scenes I had a difficulties to understand it, so I gave up and added the function to my app.

Hope you can find time to add this to the runtime...

**Comment by domderen** [comment]
Created a PR that adds that functionality into the `offline-plugin`: https://github.com/NekR/offline-plugin/pull/169

**Comment by NekR** [comment]
Yeah, landed in #169, not sure why this isn't closed already.
